internal/os/os.go

```diff
@@ -0,0 +1,45 @@
+package os
+
+import (
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/osbuild/osbuild-composer/internal/pipeline"
+)
+
+type OS interface {
+	// Returns a sorted list of the output formats this OS supports.
+	ListOutputFormats() []string
+
+	// Returns the canonical filename and MIME type for a given output
+	// format
+	FilenameFromType(outputFormat string) (string, string, error)
+
+	// Returns an osbuild pipeline that generates an image in the given
+	// output format with all packages and customizations specified in the
+	// given blueprint. `outputFormat` must be one returned by
+	// ListOutputFormats().
+	Pipeline(b *blueprint.Blueprint, outputFormat string) (*pipeline.Pipeline, error)
+}
+
+// An InvalidOutputFormatError is returned when a requested output format is
+// not supported. The requested format is included as the error message.
+type InvalidOutputFormatError struct {
+	Format string
+}
+
+func (e *InvalidOutputFormatError) Error() string {
+	return e.Format
+}
+
+var registered = map[string]OS{}
+
+func New(name string) OS {
+	return registered[name]
```

Comment:
I would rather put existence check with panic here.

Reply:
I thought panicking is a bit harsh, in case a caller would like to find out whether an OS exists. However, I never check for `nil` anywhere this is being used. Also, if we do have such a dynamically added OSs, we'd also need a function to list them.

Since this is not public API, we can think about all that later :) I'll add a panic now. Thanks for calling this out!

Reply:
Oh, it returns `nil` when it doesn't exist. I have to remember that interface is always pointer! :) This is kinda nonintuitive in Go. I guess returning `nil` is good. :)

Reply:
I've added the `panic` anyway, if only to cause less confusion in the near future :)